test/atomics_cxx/CMakeLists.txt

```diff
@@ -12,5 +12,6 @@ foreach(TEST ${TESTS})
     get_filename_component(TEST_NAME ${TEST} NAME_WE)
     add_executable(${TEST_NAME} ${TEST})
     target_link_libraries(${TEST_NAME} vatomic)
+    target_compile_options(${TEST_NAME} PUBLIC -O2)
```

Comment:
Don't you prefer using the CMAKE_BUILD_TYPE to create tests with different optimization levels?